src/java/htsjdk/variant/variantcontext/CommonInfo.java

```diff
@@ -247,37 +284,78 @@ public Object getAttribute(String key, Object defaultValue) {
         as a java.util.List if the value is a List or an Array,
         as a Collections.singletonList if there is only one value */
     @SuppressWarnings("unchecked")
-    public List<Object> getAttributeAsList(String key) {
+    public List<Object> getAttributeAsList(final String key) {
         Object o = getAttribute(key);
         if ( o == null ) return Collections.emptyList();
         if ( o instanceof List ) return (List<Object>)o;
         if ( o.getClass().isArray() ) return Arrays.asList((Object[])o);
         return Collections.singletonList(o);
     }
 
-    public String getAttributeAsString(String key, String defaultValue) {
-        Object x = getAttribute(key);
+    /**
+     * return an attribute as a String.
+     * if given key is not found the defaultValue is returned.
+     * if the value is not a String <code> String.valueOf(x)</code> is returned.
+     * 
+     * @param key the attribute key
+     * @param defaultValue  the default value
+     * @return the attribute as a String
+     */
+    public String getAttributeAsString(final String key, final String defaultValue) {
+        final Object x = getAttribute(key);
         if ( x == null ) return defaultValue;
         if ( x instanceof String ) return (String)x;
         return String.valueOf(x); // throws an exception if this isn't a string
```

Comment:
This comment "throws an exception if this isn't a string" isn't right -- `String.valueOf(x)` calls `x.toString()`, regardless of whether x is already a String or not.